ui.apps/src/main/content/jcr_root/apps/core/cif/components/commerce/product/v3/product/clientlib/js/gallery.js

```diff
@@ -0,0 +1,213 @@
+        if (currentlySelectedThumb) {
+            currentlySelectedThumb.classList.remove('thumbnail__rootSelected');
+        }
+        const currentThumb = this._rootNode.querySelector(`button[data-gallery-index='${index}']`);
+        currentThumb.classList.add('thumbnail__rootSelected');
+    }
+
+    /**
+     * Adds the listeners to the thumbnails
+     * @private
+     */
+    _installThumbnailEvents() {
+        const handleThumbnailClick = (idx, event) => {
+            let currentTarget = event.currentTarget;
+            this._switchCurrentImage(idx);
+        };
+
+        document.querySelectorAll(Gallery.selectors.galleryThumbnail).forEach((node, idx) => {
+            node.addEventListener('click', event => {
+                event.preventDefault();
+                handleThumbnailClick(idx, event);
+            });
+        });
+    }
+
+    /**
+     * Installs events listeners
+     * @private
+     */
+    _installEvents() {
+        this._installThumbnailEvents();
+        document
+            .querySelector(Gallery.selectors.leftArrow)
+            .addEventListener('click', event => this._handleArrowClick(event));
+        document
+            .querySelector(Gallery.selectors.rightArrow)
+            .addEventListener('click', event => this._handleArrowClick(event));
+    }
+
+    /**
+     * Handles the click on the "chevrons" for gallery navigation
+     * @param event
+     * @private
+     */
+    _handleArrowClick(event) {
+        event.preventDefault();
+        const direction = event.currentTarget.dataset['galleryRole'];
+
+        if (direction === 'moveleft') {
+            if (this._currentItemIndex <= 0) {
+                this._currentItemIndex = this._galleryItems.length - 1;
+            } else {
+                this._currentItemIndex--;
+            }
+        } else if (direction === 'moveright') {
+            if (this._currentItemIndex >= this._galleryItems.length - 1) {
+                this._currentItemIndex = 0;
+            } else {
+                this._currentItemIndex++;
+            }
+        }
+
+        this._switchCurrentImage(this._currentItemIndex);
+    }
+}
+
+Gallery.selectors = {
+    galleryRoot: "div[data-gallery-role='galleryroot']",
+    currentImageContainer: "img[data-gallery-role='currentimage']",
+    galleryThumbnail: "button[data-gallery-role='galleryitem']",
+    thumbnailRoot: 'div.thumbnailList__root',
+    leftArrow: "button[data-gallery-role='moveleft']",
+    rightArrow: "button[data-gallery-role='moveright']"
+};
+
+Gallery.events = {
+    variantChanged: 'variantchanged'
+};
+
+(function(document) {
+    function onDocumentReady() {
+        const galleryRoot = document.querySelector(Gallery.selectors.galleryRoot);
+
+        const galleryItems = galleryRoot ? JSON.parse(galleryRoot.dataset.galleryItems) : { assets: [] };
+        const gallery = new Gallery({ galleryItems });
```

Comment:
Could this be conditional as well? Only if `galleryRoot` exists, we create a new `Gallery` instance.